yarn-project/ethereum/src/l1_contract_addresses.ts

```diff
@@ -0,0 +1,82 @@
+import { EthAddress } from '@aztec/foundation/eth-address';
+
+/**
+ * Provides the directory of current L1 contract addresses
+ */
+export class L1ContractAddresses {
+  constructor(
+    /**
+     * Rollup Address.
+     */
+    public rollupAddress?: EthAddress,
+    /**
+     * Registry Address.
+     */
+    public registryAddress?: EthAddress,
+    /**
+     * Inbox Address.
+     */
+    public inboxAddress?: EthAddress,
+    /**
+     * Outbox Address.
+     */
+    public outboxAddress?: EthAddress,
+    /**
+     * Data Emitter Address.
+     */
+    public contractDeploymentEmitterAddress?: EthAddress,
+    /**
+     * Decoder Helper Address.
+     */
+    public decoderHelperAddress?: EthAddress,
```

Comment:
I'd try to keep these as required and not optional. All through the code we now need non-null assertions (exclamation marks), so we lose any help from the compiler on detecting where we may have forgotten initialising them.

Searching through this PR, the only place where I found that we need them to be optional is in `create(P2P)Sandbox`, where we initialise a new `AztecNodeConfig`. I think we should be able to delay initialising the config until we have all required data, so we can keep these as required.

So instead of:
```ts
export async function createP2PSandbox() {
  const aztecNodeConfig: AztecNodeConfig = { ...getConfigEnvVars() };
  const rpcConfig = getRpcConfigEnvVars();
  const privateKeyAccount = privateKeyToAccount(aztecNodeConfig.publisherPrivateKey);

  const l1Contracts = await waitThenDeploy(aztecNodeConfig, () =>
    retrieveL1Contracts(aztecNodeConfig, privateKeyAccount),
  );
  aztecNodeConfig.l1Contracts.rollupAddress = l1Contracts.l1ContractAddresses.rollupAddress;
  aztecNodeConfig.l1Contracts.contractDeploymentEmitterAddress =
    l1Contracts.l1ContractAddresses.contractDeploymentEmitterAddress;
  aztecNodeConfig.l1Contracts.inboxAddress = l1Contracts.l1ContractAddresses.inboxAddress;
  aztecNodeConfig.l1Contracts.registryAddress = l1Contracts.l1ContractAddresses.registryAddress;

  const node = await AztecNodeService.createAndSync(aztecNodeConfig);
```

We can try something like:
```ts
export async function createP2PSandbox() {
  const rpcConfig = getRpcConfigEnvVars();
  const privateKeyAccount = privateKeyToAccount(aztecNodeConfig.publisherPrivateKey);

  const l1Contracts = await waitThenDeploy(aztecNodeConfig, () =>
    retrieveL1Contracts(aztecNodeConfig, privateKeyAccount),
  );
  const aztecNodeConfig: AztecNodeConfig = { 
      ...getConfigEnvVars(),
      l1Contracts: l1Contracts,
   };

  const node = await AztecNodeService.createAndSync(aztecNodeConfig);
```